helm/prometheus-rules/templates/alerting-rules/prometheus.rules.yml

```diff
@@ -65,3 +65,17 @@ spec:
         team: atlas
         topic: observability
         cancel_if_outside_working_hours: "true"
+    ## Page Atlas if prometheus agent fails to send samples to MC prometheus.
+    - alert: PrometheusAgentRemoteWriteNotWorking
```

Comment:
This rule is supposed to run on all clusters? Even MCs? Do they have an agent too?

Reply:
This alert will be triggered on MC

Reply:
If the metrics of agent is absent, it means that there's an issue within remotewrite or prometheus

Reply:
can you check now?

Reply:
Sorry, dumb me again:
I like the new query (`absent_over_time`). But, as far as I understand, this query will run on all prometheis, WCs and MCs. Are they all supposed to have a prometheus-agent running?

I tested on a MC (gorilla) and on a WC (gorilla/zj88t), and the query would page on both.

Reply:
We can use a label to ensure this is only running on WC promethues using this label:
https://github.com/giantswarm/prometheus-rules/blob/0ae0e62c1aba17298f960877917ebaa703637a43/helm/prometheus-rules/templates/alerting-rules/app.rules.yml#L7 with the value of workload_cluster.

You need to move the alert to a new file. Something like prometheus-agent.rules.yaml for example

Reply:
All WCs running version 19+ should have the agent but we have no way to know :(

Reply:
Unless we can use some kind of join with a metric about cluster versions

Reply:
Actually, that's what this label is for. Only thé wc prometheus in the wc will pick up this rule. I'm not taking about the label of thé metric but thé label of thé prometheus rule CR. It's set in the prometheusrule selector